stable/nginx-ingress/templates/default-backend-deployment.yaml

```diff
@@ -10,6 +10,14 @@ metadata:
     release: {{ .Release.Name }}
   name: {{ template "nginx-ingress.defaultBackend.fullname" . }}
 spec:
+  selector:
+    matchLabels:
+      app: {{ template "nginx-ingress.name" . }}
+      component: "{{ .Values.defaultBackend.name }}"
+      release: {{ .Release.Name }}
+      {{- if .Values.defaultBackend.podLabels }}
+{{ toYaml .Values.defaultBackend.podLabels | indent 6 }}
```

Comment:
Same as previous comment.

> Let's not add the podLabels to the selector since this field is immutable.

Reply:
According to [https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector)

> `.spec.selector` is an required field that specifies a label selector for the Pods targeted by this Deployment.
>
> `.spec.selector` must match `.spec.template.metadata.labels`, or it will be rejected by the API.
> 
> In API version `apps/v1`, `.spec.selector` and `.metadata.labels` do not default to `.spec.template.metadata.labels` if not set. So they must be set explicitly. Also note that `.spec.selector` is immutable after creation of the Deployment in `apps/v1`.

Reply:
Since the field is immutable we want to only use the minimum necessary amount of labels to uniquely identify the pods. Otherwise, there is a large potential to not be able to do upgrades. For instance, if I add a `version` label, then I will always need to re-create the deployment in order to upgrade.